main_test.go

```diff
@@ -108,15 +108,22 @@ func TestCompiler(t *testing.T) {
 	}
 }
 
+var additionalArgs = map[string][]string{
+	"testdata/wasi_env.go": {"--env", "ENV1=VALUE1", "--env", "ENV2=VALUE2"},
+}
+
 func runPlatTests(target string, matches []string, t *testing.T) {
 	t.Parallel()
 
 	for _, path := range matches {
-		path := path // redefine to avoid race condition
+		if strings.Contains(path, "wasi") && target != "wasi" {
```

Comment:
I think you can simplify this by calling `runTest("testdata/wasi_env.go", "wasi", t, {"--env", "ENV1=VALUE1", "--env", "ENV2=VALUE2"})` directly in the loop above:

```go
		t.Run("WASI", func(t *testing.T) {
			runPlatTests("wasi", matches, t)
			runTest("testdata/wasi_env.go", "wasi", t, {"--env", "ENV1=VALUE1", "--env", "ENV2=VALUE2"})
		})
```